src/geometric_utils.ts

```diff
@@ -202,31 +228,40 @@ export class GeometricUtils {
     }
 
     // Subtract poly2 from poly1. Result is a complex polygon ULabelSpatialPayload2D[], with any holes preserved.
-    public static subtract_polygons(complex_poly1: ULabelSpatialPayload2D[], complex_poly2: ULabelSpatialPayload2D[]): ULabelSpatialPayload2D[] {
+    public static subtract_polygons(
+        complex_poly1: ULabelSpatialPayload2D[],
+        complex_poly2: ULabelSpatialPayload2D[],
+    ): ULabelSpatialPayload2D[] {
         let ret: ULabelSpatialPayload2D[];
         complex_poly1 = GeometricUtils.ensure_valid_turf_complex_polygon(complex_poly1);
         complex_poly2 = GeometricUtils.ensure_valid_turf_complex_polygon(complex_poly2);
-        let temp = turf.difference(turf.polygon(complex_poly1), turf.polygon(complex_poly2));
+
+        const temp = turf.difference(
+            turf.polygon(complex_poly1),
+            turf.polygon(complex_poly2),
+        );
         // when temp is null, return
         if (temp === null) {
             return null;
-        } else {
-            temp = temp.geometry.coordinates;
         }
+        const temp_coords = <ULabelSpatialPayload2D[]>temp.geometry.coordinates;
 
         // When turf.difference creates a fill, it adds it as a new polygon, ie [complex_poly, fill] instead of just complex_poly
         // so we need to append the fill to the complex_poly and return that when turf returns a quad nested list instead of a triple nested list
-        if (temp[0][0][0][0] === undefined) {
-            ret = temp;
+        if (temp_coords[0][0][0][0] === undefined) {
+            ret = temp_coords;
         } else {
-            ret = temp[0].concat(temp[1]);
+            // TODO (joshua-dean): See if this casting can be better
+            ret = <ULabelSpatialPayload2D[]><unknown> temp_coords[0].concat(temp[1]);
         }
```

Comment:
This is the ugliest of the bunch. I added another temp variable and then did an `<unknown>` cast as recommended by the rule making the complaint. I don't like it, but I don't think this breaks anything either